protocol/0087-VAMM-automated_market_maker.md

```diff
@@ -0,0 +1,94 @@
+
+An automated market maker is configured at a per-key level, and is enabled by submitting a transaction with the requisite parameters. At this point in time the protocol will move committed funds to a sub-account which will be used to manage order and position margin for the AMM. Once enabled, the configuration will be queried once per block and the resultant orders will be placed on the book, combined with all current orders from the key being cancelled. 
+
+Each party may have only one AMM configuration per market.
+
+## Process Overview
+
+The configuration and resultant lifecycle of an automated market maker is as follows:
+
+- Party funds a key which will be used by the strategy with desired token amounts.
+- Party submits a transaction containing configuration for the strategy on a given market. This will contain:
+  - Amount of funds to commit
+  - Price bounds (upper, lower, base)
+  - Granularity of price levels to post
+- Once accepted, the network will transfer funds to a sub-account and use the other parameters for maintaining the position.
+- At each block, immediately prior to the evaluation of liquidity provision for SLA purposes, for each configured AMM:
+  - The party's available balance (including margin and general accounts) for trading on the market will be checked. If the total balance is `0` the AMM configuration will be cancelled. 
+  - Each running AMM will be queried for it's provided orders. All orders from this party on this market will then be cancelled, followed by the placement of these new orders on the book. There are a couple of things to note here:
+    - All orders are placed as-if arriving from the party's key externally.
+- If the party submits a `CancelAMM` transaction the AMM configuration for that party, on that market, will be cancelled. All active orders from the AMM will be cancelled and all funds and positions associated with the sub-account will be transferred back to the main account.
+
+## Sub-Account Configuration
+
+Each main Vega key will have one associated sub account for a given market, on which an AMM may be set up. The account key should be generated through a hash of the main account key plus the ID of the market to generate a valid Vega address in a predictable manner. Outside of the AMM framework the sub-accounts are treated identically to any other account, they will have the standard associated margin/general/bond accounts and be able to place orders if required as with any other account. The key differentiator is that no external party will have the private key to control these accounts directly. The maintenance of such an account will be performed through a few actions:
```

Comment:
It may be nice if I could control the account at least for purpose of posting market orders(*) to reduce position - but that's probably an unnecessary level of complexity. 

(*) of course this can only help if there is other volume on the book not just my amm volume

Reply:
Would have to change the way the bot works entirely for that to make sense though. A uniswap-esque liquidity pool's price is ultimately determined by it's position, so reducing position would just change it's idea of what the price should be and it would be back where it was seconds later :D. One proviso though is what should happen when someone changes the bounds, as then they probably do want to change position more aggressively (at the least with not post-only orders)